src/rulesets/oas/functions/oasUnusedComponent.ts

```diff
@@ -0,0 +1,43 @@
+import type { IFunction, IFunctionContext, IFunctionResult } from '../../../types';
+
+export const oasUnusedComponent: IFunction<{}> = function (
+  this: IFunctionContext,
+  targetVal,
+  opts,
+  _paths,
+  otherValues,
+) {
+  const results: IFunctionResult[] = [];
+
+  if (targetVal.components === void 0) {
```

Comment:
How about utilizing `isObject` util here? Just in case one provides null, or a number, or boolean, or whatever else.
You never know.
```suggestion
  if (!isObject(targetVal.components)) {
```
I don't recall where exactly we have it implemented, but other fns do use it, so you should be able to find it pretty easily.

Reply:
I've made this change.